plugin/pkg/proxy/connect.go

```diff
@@ -148,3 +182,22 @@ func (p *Proxy) Connect(ctx context.Context, state request.Request, opts Options
 }
 
 const cumulativeAvgWeight = 4
+
+// Function to return an empty response with TC (truncated) bit set.
+// This is to mitigate the effects of an upstream server sending UDP responses that exceed the maximum size of 512 bytes.
+func truncateResponse(response *dns.Msg) *dns.Msg {
+
+	truncatedResponse := response.Copy()
+
+	// Clear out Answer, Extra, and AuthenticatedData sections
+	truncatedResponse.Answer = nil
+	truncatedResponse.Extra = nil
+
```

Comment:
Also need to clear out the Authority section (it's called Ns in the msg struct).